### PR TITLE
Move lazy static values to heap

### PR DESCRIPTION
**Pull request by jules**
# What ❔

Moves lazy static consts to the heap.

## Why ❔

There were reports of stack overflows on certain hardware.

## Checklist

- [x] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [x] Tests for the changes have been added / updated.
- [x] Documentation comments have been added / updated.
- [x] Code has been formatted via `zk fmt` and `zk lint`.

### CODE REVIEW

File: src/kzg/mod.rs [hunk 1]
```diff
@@ -96,9 +98,9 @@ lazy_static::lazy_static! {
         // cooley-tukey algorithm. the roots need to be inverted, and all results need to be
         // divided by 2^12.
         let roots = ROOTS_OF_UNITY
-            .into_iter()
+            .iter()
             .take(FIELD_ELEMENTS_PER_BLOB / 2)
-            .map(|r| if r != Fr::one() { r.inverse().unwrap() } else { r })
+            .map(|r| if *r != Fr::one() { r.inverse().unwrap() } else { *r })
```
Comment (shamatar) [inline src/kzg/mod.rs]: There is no point in special casing here

File: src/kzg/mod.rs [hunk 2]
```diff
@@ -48,7 +48,7 @@ fn hex_to_bytes(hex_string: &str) -> Vec<u8> {
 }
 
 lazy_static::lazy_static! {
-    static ref ROOTS_OF_UNITY: [Fr; FIELD_ELEMENTS_PER_BLOB] = {
+    static ref ROOTS_OF_UNITY: Vec<Fr> = {
```
Comment (shamatar) [inline src/kzg/mod.rs]: Either `Box<[Fr; 4096]>` or `Box<[Fr]>`, you never need static immutable vector

File: src/kzg/mod.rs [hunk 3]
```diff
@@ -188,8 +190,9 @@ pub fn compute_proof(blob: &[Fr], z: &Fr) -> (G1Affine, Fr) {
         })
         .collect::<Vec<Fr>>();
     let denom_poly = ROOTS_OF_UNITY_BRP
-        .into_iter()
-        .map(|mut el| {
+        .iter()
+        .map(|el| {
+            let mut el = el.clone();
```
Comment (shamatar) [inline src/kzg/mod.rs]: it's copiable, so `iter().copied()` or just `let el = *el` is sufficient

**Review by shamatar — CHANGES_REQUESTED** [review]

**Comment by koloz193** [comment]
closing as changes were covered in https://github.com/matter-labs/era-zkevm_test_harness/pull/77

**Pull request closed without merging**